mkfs/mkfs.c

```diff
@@ -17,13 +17,36 @@
 #include <errno.h>
 #include <math.h>
 #include <locale.h>
+#include <time.h>
 
 #include "exfat_ondisk.h"
 #include "libexfat.h"
 #include "mkfs.h"
 
 struct exfat_mkfs_info finfo;
 
+/* random serial generator based on current date & time */
+static unsigned int get_new_serial(void)
+{
+	struct tm *tm;
+	struct timespec ts;
+	time_t now;
+	unsigned int x;
+	unsigned int nsec_u, nsec_l;
+
+	clock_gettime(CLOCK_REALTIME, &ts);
```

Comment:
need to check return value of clock_gettime(). If one of two(clock_gettime() or localtime()) return error, should we calculate with a fixed time?

Reply:
Will be done for clock_gettime().
Any suggestion for the *fixed time* ?

Reply:
fixed time => 1970.01.01 00:00:00 ?

Reply:
It would be converted as 0 and serial as 0x00000000.
It would be ok but how about 1583130092 ?
This is from below.
-----
commit 1acf1a564b6034b5af1e7fb23cb98cb3bb4f6003
Author: Namjae Jeon <namjae.jeon@samsung.com>
Date:   Mon Mar 2 15:21:32 2020 +0900

    exfat: add in-memory and on-disk structures and headers